Resfeber/API/TeleportAPIManager.swift

```diff
@@ -0,0 +1,297 @@
+//
+//  TeleportAPIManager.swift
+//  Resfeber
+//
+//  Created by Joshua Rutkowski on 11/13/20.
+//  Copyright © 2020 Spencer Curtis. All rights reserved.
+//
+
+import Foundation
+
+class TeleportAPIManager {
+    /// The Base URL for connecting with the Teleport json endpoint
+    let baseURL = URL(string: "https://api.teleport.org/api")!
+    
+    /// The default Headers for connecting with the Teleport json endpoint
+    var defaultHeaders = [
+        "Accept" : "application/vnd.teleport.v1+json",
+        "Content-Type" : "application/json; charset=utf-8"
+    ]
+    
+    // MARK: - Endpoints
+    /// The endpoints used to connect to the Teleport API
+    enum Endpoints {
+        case citybyName(String)
+        case citiesByID(String)
+        case citiesAlternateNames(String)
+        case urbanAreas
+        case urbanAreasByID(String)
+        case urbanAreasCities(String)
+        case urbanAreasDetails(String)
+        case urbanAreasImages(String)
+        case urbanAreasScores(String)
+        case locationsByCoordinates(String)
+    }
+    
+    // MARK: - Methods
+    
+    /// Method to search for a city by name. Returns matching cities and endpoints.
+    /// - Parameters:
+    ///   - name: Name of city
+    ///   - session: URLSession
+    private func searchCity(byName name: String, using session: URLSession = .shared) {
+        session.request(TeleportAPIManager.Endpoints.citybyName(name)) { data, response, error  in
+            
+            if error != nil || data == nil {
+                print("Client error: \(String(describing: error?.localizedDescription))")
+                return
+            }
```

Comment:
I would probably handle error and data separately.
```
if let error = error {
    NSLog("Client error: %@", error)
    return
}

guard let data = data else {
  NSLog("No data returned from the network request")
  return
}
```

That way you won't have to force unwrap your data below, and you can be more specific about what the actual error was